assets/js/wc-gateway-ppec-smart-payment-buttons.js

```diff
@@ -115,12 +88,16 @@
 							.attr( 'value', fromCheckout ? 'yes' : 'no' )
 						)
 						.serialize();
-
-					return paypal.request( {
+					
+					return fetch( wc_ppec_context.start_checkout_url, {
```

Comment:
I don't think this will work on Internet Explorer 11 p7bbVw-3nP-p2

Reply:
You're right; it will not. The [documentation provided by PayPal](https://developer.paypal.com/docs/checkout/reference/upgrade-integration/#6-fix-deprecations) does say to use `fetch` instead of their `paypal.request` function, although I don't know whether the `paypal.request` function is still included.

I guess that means that PayPal doesn't include support for IE11 anymore? In any case, we could use a transpiler like babel [[#](https://heyhydra.wordpress.com/2019/10/02/minimum-browser-requirements/#comment-8364)] [[#](https://babeljs.io/)] (or webpack I guess?) to be able to use the newer functions. There are even 3rd party libraries [[#](http://bluebirdjs.com/docs/getting-started.html)] made specifically to replicate this functionality and make it available in older browsers.

That said, I don't know what the best way to handle this is. As much as I'd like to say "just don't support IE11", that may not really be a viable option since IE11 doesn't have an official end of support date.

>Internet Explorer 11 will continue receiving security updates and technical support for the lifecycle of the version of Windows on which it is installed.

[Source](https://support.microsoft.com/en-us/help/17454/lifecycle-faq-internet-explorer).

Reply:
I wouldn't say PayPal doesn't support it, since the doc there says "Use the built-in browser Promise, with a polyfill or your promise library of choice." The transpilation via Babel would just involve the inclusion of a polyfill library ([core-js](https://github.com/zloirock/core-js)), though `fetch` isn't included so we would need [another polyfill](https://github.com/github/fetch) for that.

Reply:
Let's plan on transpiling then. This PR just got bigger. I think it might be better to tackle other issues first and come back to this one when I can dedicate a couple people to it for a longer time period. What do y'all think?

Reply:
In case it wasn't clear, no transpilation is necessary (since this isn't a matter of new syntax), but we'd need to include polyfills for `Promise` and `window.fetch`.

Reply:
>I think it might be better to tackle other issues first and come back to this one when I can dedicate a couple people to it for a longer time period.

Definitely, especially when we take https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/622#discussion_r331124820 into account. I do think we can clear some low-ish hanging fruit here:
- We should be able to migrate all the configuration flags with relatively little effort: 
    - `commit`
    - `disable-funding`
    - `disable-card`
- Add polyfills as suggested by @dechov ^

That said, it will still take some time (maybe 0.5 to 1 day of effort? More if there are any unexpected complications) so @allendav I'd like your opinion on whether to leave this PR until we can focus on it, or finish the items above first.

Reply:
Sorry for the delay getting back here. We'll circle back to this PR for the 1.7.0 milestone as part of Sprint 6